Gems/EMotionFX/Code/Source/Editor/Plugins/Ragdoll/JointSwingLimitManipulators.cpp

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzFramework/Physics/Configuration/JointConfiguration.h>
+#include <AzFramework/Viewport/ViewportColors.h>
+#include <AzFramework/Viewport/ViewportConstants.h>
+#include <AzToolsFramework/Manipulators/ManipulatorManager.h>
+#include <AzToolsFramework/Manipulators/ManipulatorView.h>
+#include <EMotionFX/CommandSystem/Source/JointLimitCommands.h>
+#include <EMotionStudio/EMStudioSDK/Source/EMStudioManager.h>
+#include <Editor/Plugins/Ragdoll/JointLimitRotationManipulators.h>
+#include <Editor/Plugins/Ragdoll/JointSwingLimitManipulators.h>
+#include <Editor/Plugins/Ragdoll/RagdollJointLimitWidget.h>
+
+namespace EMotionFX
+{
+    static constexpr float ManipulatorOffsetX = 0.2f; // manipulator position offset along the joint parent frame's X axis
+    static constexpr float ManipulatorScale = 400.0f; // scaling factor between linear position of manipulator and swing limit in degrees
+    static constexpr float ManipulatorInverseScale = 1.0f / ManipulatorScale;
+
+    JointSwingLimitManipulators::JointSwingLimitManipulators()
+    {
+        m_adjustJointLimitCallback = AZStd::make_unique<PhysicsSetupManipulatorCommandCallback>(this, false);
+        EMStudio::GetCommandManager()->RegisterCommandCallback("AdjustJointLimit", m_adjustJointLimitCallback.get());
+    }
+
+    JointSwingLimitManipulators::~JointSwingLimitManipulators()
+    {
+        EMStudio::GetCommandManager()->RemoveCommandCallback(m_adjustJointLimitCallback.get(), false);
+    }
+
+    void JointSwingLimitManipulators::Setup(PhysicsSetupManipulatorData& physicsSetupManipulatorData)
+    {
+        m_physicsSetupManipulatorData = physicsSetupManipulatorData;
+
+        if (!m_physicsSetupManipulatorData.HasJointLimit())
+        {
+            return;
+        }
+
+        AzFramework::DebugDisplayRequestBus::BusPtr debugDisplayBus;
+        AzFramework::DebugDisplayRequestBus::Bind(debugDisplayBus, m_viewportId);
+        m_debugDisplay = AzFramework::DebugDisplayRequestBus::FindFirstHandler(debugDisplayBus);
```

Comment:
question: This is an interesting way of doing this, usually we just listen to a draw notification bus which passes through the debugDisplayBus, is there a reason why we can't do this in this case? Not a huge deal just curious

Reply:
We discussed this offline. I'm not sure the way it works in the editor is directly applicable here right now. We can look into adding it later.

The pattern I am using here is quite common already in the EMFX gem.